bundle-workflow/src/manifests/build_manifest.py

```diff
@@ -96,6 +96,12 @@ def __to_dict__(self):
             ),
         }
 
+    def get_component(self, component_name):
+        for component in self.components:
+            if component.name == component_name:
+                return component
+        raise ComponentNotFoundError(f"{component_name} not found in build manifest.yml")
```

Comment:
This is OK, but consider two improvements:

1. I'd want to be able to write `manifest.components['name']`. 
2. You can use `next(...)` iterators in Python making this a lot cleaner.

Reply:
2 is addressed. 1 can be taken up later as a refactoring improvement